### PR TITLE
Issue 232 external date

### PR DESCRIPTION
**Pull request by dougli1sqrd**
This let's the project yaml have a property called `use_external_date`, which defaults to False. 

If True, when an ontology project is created and templated, the `run.sh` will set the docker environment with 
```
TODAY=`date +%Y-%m-%d`
```
when running commands.

`TODAY` is then picked up by the Makefile in `src/ontology` to properly set the date and Version URI.

### CODE REVIEW

**Comment by dougli1sqrd** [comment]
For https://github.com/INCATools/ontology-development-kit/issues/232

File: odk/odk.py [hunk 1]
```diff
@@ -50,20 +50,20 @@ class Product(JsonSchemaMixin):
 
     Here, a product is something that is produced by an ontology workflow.
     A product can be manifested in different formats.
-
```
Comment (cmungall) [inline odk/odk.py]: It would be good to avoid spurious diffs. Do you want to write guidelines and add a linter to travis? How do other projects do this?

**Comment by cmungall** [comment]
It's hard to evaluate due to the spurious diffs but I think this is makes sense. Also spurious diffs can cause merge conflicts, not sure if Nico has any open. How hard is it to redo PR with only meaningful changes?

File: odk/odk.py (hunk 1 above)
Comment (dougli1sqrd) [reply odk/odk.py]: Ah yeah, sorry about that. I can try and fix my editor from changing whitespace like that. I have no opinion really as to "correct" white space. This is just something my editor does to everyone's annoyance. I don't think we need a linter or anything.

**Comment by dougli1sqrd** [comment]
I can try and redo it. Sorry, that's just my editor being weird. I don't know why it does it, but I'm sure it's a setting I can turn off.

**Comment by dougli1sqrd** [comment]
okay white space should be fixed.

**Comment by matentzn** [comment]
Some whitespaces are still there. And I learned something new ?= in the makefile! Awesome.

**Pull request merged**